src/framework/handler/core.clj

```diff
@@ -24,24 +24,20 @@
   [deps]
   (fn handle*
     ([http-request]
-     (if (:websocket? http-request)
-       (let [state (state/make deps http-request)
-             queue (list #(interceptor.queue/execute % (:router-interceptors deps))
-                         #(route/match %)
-                         #(interceptor.queue/execute % (:web-socket-interceptors deps)))
-             flow (-> (xiana/apply-flow-> state queue)
-                      (xiana/extract))
-             channel (get-in flow [:response-data :channel])]
-         (if channel
-           (as-channel http-request channel)
-           (:response flow)))
-       (let [state (state/make deps http-request)
-             queue (list #(interceptor.queue/execute % (:router-interceptors deps))
-                         #(route/match %)
-                         #(interceptor.queue/execute % (:controller-interceptors deps)))]
-         (-> (xiana/apply-flow-> state queue)
-             (xiana/extract)
-             :response))))
-
+     (try (let [websocket? (:websocket? http-request)
+                state (state/make deps http-request)
+                queue (list #(interceptor.queue/execute % (:router-interceptors deps))
+                            #(route/match %)
+                            #(interceptor.queue/execute % (if websocket?
+                                                            (:web-socket-interceptors deps)
+                                                            (:controller-interceptors deps))))
+                result (-> (xiana/apply-flow-> state queue)
+                           (xiana/extract))
+                channel (get-in result [:response-data :channel])]
+            (if (and websocket? channel)
+              (as-channel http-request channel)
+              (:response result)))
+          (catch Exception e (or (ex-data e)
```

Comment:
Wouldn't it be better to handle exceptions in an interceptor?

Reply:
All interceptors wrapped into a try-catch, only the action Fns are handled. In theory we can get rid of this wrapping.